src/SV0001Itinerary/Utils/api/DB0002Itineraries.ts

```diff
@@ -0,0 +1,45 @@
+import db from '../fireB/firestore'
+import { doc, collection, getDoc, setDoc, addDoc, deleteDoc, getDocs } from 'firebase/firestore';
+import type { DB0002ItineraryType } from 'SV0001Itinerary/Utils/types/DB0002ItineraryType';
+import { initItinerary } from 'SV0001Itinerary/Utils/types/DB0002ItineraryType';
+
+export const collectionName = 'Itineraries';
+const collectionRef = collection(db, collectionName);
+
+export const read = async (id: string): Promise<DB0002ItineraryType | null> => {
+    try {
+        let docSnap = await getDoc(doc(collectionRef, id));
+        const itinerary: DB0002ItineraryType = {
+            id: id,
+            title: docSnap.data()?.title,
+        };
+        return itinerary;
+    } catch {
+        return null;
+    }
+}
+
+export const readAll = async (): Promise<DB0002ItineraryType[]> => {
+    const querySnapshot = await getDocs(collectionRef);
+    return (querySnapshot.docs.map((doc) => {
+        const itinerary: DB0002ItineraryType = {
+            id: doc.data().id,
+            title: doc.data().title,
+        };
+        return itinerary;
+    }));
```

Comment:
↓のように出来ないかな（違和感あればご意見ほしい）
```suggestion
    return (querySnapshot.docs.map((doc): DB0002ItineraryType =>
        doc.data();
     ));
```

Reply:
こっちの方がわかりやすいわ、カラムの数増えた時に
前の記述だと修正だるいから